src/wrapper.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * @author acvetkov@yandex-team.ru
+ * @overview
+ */
+
+'use strict';
+
+import {get} from './object';
+
+/**
+ * chrome api object
+ * @see https://developer.chrome.com/extensions/api_index
+ */
+const chromeApi = getChromeApi();
+
+/**
+ * Create getters for async methods
+ * @param {Object} obj
+ * @param {String} namespace
+ * @param {Array<String>} asyncMethods
+ * @param {Function} Thenable
+ * @returns {Object}
+ */
+export function wrapAsyncMethods(obj, namespace, asyncMethods = [], Thenable) {
+    return asyncMethods.reduce((result, method) => {
+        return appendAsyncMethod(obj, namespace, method, Thenable);
+    }, obj);
+}
+
+/**
+ * Wrap sync methods to promise
+ * @param {Object} obj
+ * @param {String} namespace
+ * @param {Array<String>} syncMethods
+ * @param {Function} Thenable
+ * @returns {*}
+ */
+export function wrapSyncMethods(obj, namespace, syncMethods = [], Thenable) {
+    return syncMethods.reduce((result, method) => {
+        return appendSyncMethod(obj, namespace, method, Thenable);
+    }, obj);
+}
+
+/**
+ * Get chrome api object
+ * @returns {chrome}
+ */
+function getChromeApi() {
+    if (typeof window !== 'undefined' && window.chrome) {
+        return window.chrome;
+    }
+    if (typeof global !== 'undefined' && global.chrome) {
```

Comment:
Instead of `Function` method `var globalObject = typeof self !== 'undefined' ? self : global` can be used. Or something like this